build break with gcc11

Getting a build break with gcc11
```
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/./larpandoracontent/LArHelpers/LArMvaHelper.h:346:27: error: loop variable 'feature' creates a copy from type 'const MvaFeature' {aka 'const lar_content::MvaTypes::InitializedDouble'} [-Werror=range-loop-construct]
  346 |     for (const MvaFeature feature : featureList)
      |                           ^~~~~~~
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/./larpandoracontent/LArHelpers/LArMvaHelper.h:346:27: note: use reference type to prevent copying
  346 |     for (const MvaFeature feature : featureList)
      |                           ^~~~~~~
      |                           &
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/./larpandoracontent/LArHelpers/LArMvaHelper.h: In instantiation of 'static pandora::StatusCode lar_content::LArMvaHelper::WriteFeaturesToFileImpl(std::ofstream&, const string&, TLIST&&) [with TLIST = std::vector<lar_content::MvaTypes::InitializedDouble>&; std::ofstream = std::basic_ofstream<char>; std::string = std::__cxx11::basic_string<char>]':
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/./larpandoracontent/LArHelpers/LArMvaHelper.h:314:5:   required from 'static pandora::StatusCode lar_content::LArMvaHelper::WriteFeaturesToFile(std::ofstream&, const string&, TLIST&&, TLISTS&& ...) [with TLIST = std::vector<lar_content::MvaTypes::InitializedDouble>&; TLISTS = {}; std::ofstream = std::basic_ofstream<char>; std::string = std::__cxx11::basic_string<char>]'
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/./larpandoracontent/LArHelpers/LArMvaHelper.h:211:5:   required from 'static pandora::StatusCode lar_content::LArMvaHelper::ProduceTrainingExample(const string&, bool, TLISTS&& ...) [with TLISTS = {std::vector<lar_content::MvaTypes::InitializedDouble, std::allocator<lar_content::MvaTypes::InitializedDouble> >&}; std::string = std::__cxx11::basic_string<char>]'
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/larpandoracontent/LArControlFlow/BdtBeamParticleIdTool.cc:119:49:   required from here
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/./larpandoracontent/LArHelpers/LArMvaHelper.h:330:27: error: loop variable 'feature' creates a copy from type 'const MvaFeature' {aka 'const lar_content::MvaTypes::InitializedDouble'} [-Werror=range-loop-construct]
  330 |     for (const MvaFeature feature : featureList)
      |                           ^~~~~~~
/tmp/user/spack-stage/spack-stage-larcontent-3.22.9-oncu73z4c6lcziylnazx7pua7iyido76/spack-src/./larpandoracontent/LArHelpers/LArMvaHelper.h:330:27: note: use reference type to prevent copying
  330 |     for (const MvaFeature feature : featureList)
      |                           ^~~~~~~
      |                           &
```

--- a/larpandoracontent/LArHelpers/LArMvaHelper.h
+++ b/larpandoracontent/LArHelpers/LArMvaHelper.h
@@ -327,7 +327,7 @@ inline pandora::StatusCode LArMvaHelper::WriteFeaturesToFile(std::ofstream &, co
 template <typename TLIST>
 pandora::StatusCode LArMvaHelper::WriteFeaturesToFileImpl(std::ofstream &outfile, const std::string &delimiter, TLIST &&featureList)
 {
-    for (const MvaFeature feature : featureList)
+    for (const MvaFeature &feature : featureList)
         outfile << feature.Get() << delimiter;
 
     return pandora::STATUS_CODE_SUCCESS;
@@ -343,7 +343,7 @@ LArMvaHelper::MvaFeatureVector LArMvaHelper::ConcatenateFeatureLists(TLIST &&fea
 
     LArMvaHelper::MvaFeatureVector featureVector;
 
-    for (const MvaFeature feature : featureList)
+    for (const MvaFeature &feature : featureList)
         featureVector.push_back(feature);
 
     LArMvaHelper::MvaFeatureVector newFeatureVector = ConcatenateFeatureLists(std::forward<TLISTS>(featureLists)...);

